design/user-id-update.md

```diff
@@ -0,0 +1,44 @@
+## Make User IDs persistent across session ID, but unique per AppID
+
+This is related to github issue #129.
+
+I want us to redo user IDs and persistence. Current implementation is inconsistent across testbed+Altspace, is inconsistent across registered/unregistered apps, and was designed this way based off of us not having a permission system.
+
+For Background for current implementation, see Issue #129:
+1. Eric wrote: "In a space where an MRE is instantiated multiple times with the same sessionId, a given user in the world will have a different userId in each instance."
+2. Steven wrote this comment : "@eanders-MS This is by design. GDPR requires us to not allow users to be tracked without their permission. As such, any given Altspace user will be assigned a different ID for each instance of a given app. I agree that a persistent ID per app would be tremendously useful, but we have to come up with a strategy to protect our users' privacy first."
+3. Eric wrote about a design:
+	"Registered apps: Same userId for a given user across multiple instances of an app (because the AppId is consistent).
+	Unregistered apps: Different userIds for a given user across multiple instances of an app (because the AppId is derived from the space component id)."
+	"We talked about adding functionality on the website for users to be able to be forgotten by all apps. How this would work: When generating the userId, include a key taken from the user record. When the user chooses to be forgotten, we would roll this key"
+
+Currently in Altspace we use: UserID= combination of altspaceUserID, sessionID, and SCID
+
```

Comment:
Actually, Altspace only uses the altspaceUserID, and the first non-null string of appID, sessionID, and SCID. AppID here is only defined for registered apps spawned from the world editor.

Reply:
I updated my design with this note.